qiskit/transpiler/passmanager.py

```diff
@@ -33,49 +38,102 @@
 class PassManager(BasePassManager):
     """Manager for a set of Passes and their scheduling during transpilation."""
 
-    PASS_RUNNER = RunningPassManager
-
     def __init__(
         self,
-        passes: PassSequence | None = None,
+        passes: Task | list[Task] = None,
         max_iteration: int = 1000,
     ):
-        """Initialize an empty `PassManager` object (with no passes scheduled).
+        """Initialize an empty pass manager object.
 
         Args:
-            passes: A pass set (as defined in :py:func:`qiskit.transpiler.PassManager.append`)
-                to be added to the pass manager schedule.
+            passes: A pass set to be added to the pass manager schedule.
             max_iteration: The maximum number of iterations the schedule will be looped if the
                 condition is not met.
         """
-        super().__init__(passes, max_iteration)
-        self.property_set = None
+        super().__init__(max_iteration=max_iteration)
+
+        # This is just for backward compatibility.
+        # RunningPassManager is now almost just an alias of FlowControllerLinear,
+        # but there is slight API difference and backward compatible RunningPassManager object
+        # must come here. Once after the API is aligned, this line will be removed.
+        self._flow_controller = RunningPassManager()
+
+        # For backward compatibility.
+        self._pass_sets = []
+        if passes:
+            self.append(passes)
+
+    def _passmanager_frontend(
+        self,
+        input_program: QuantumCircuit,
+        **kwargs,
+    ) -> DAGCircuit:
+        passmanager_ir = circuit_to_dag(input_program, copy_operations=False)
+        del input_program
 
+        return passmanager_ir
+
+    def _passmanager_backend(
+        self,
+        passmanager_ir: DAGCircuit,
+        in_program: QuantumCircuit,
+        **kwargs,
+    ) -> QuantumCircuit:
+        out_program = dag_to_circuit(passmanager_ir)
+
+        out_name = kwargs.get("output_name", None)
+        if out_name:
+            out_program.name = out_name
```

Comment:
```suggestion
        out_name = kwargs.get("output_name", None)
        if out_name is not None:
            out_program.name = out_name
```

subtle, but if `output_name` was `""` then it wouldn't get set, but that's not the same as it being absent.

Reply:
Done in [f0a547a](https://github.com/Qiskit/qiskit/pull/10127/commits/f0a547a609378238ea32827418a810588cdbd522)